src/Tools/dotnet-dump/Analyzer.cs

```diff
@@ -105,7 +105,7 @@ public Task<int> Analyze(FileInfo dump_path, string[] command)
                         }
                     }
                 }
-                if (!_consoleProvider.Shutdown)
+                if (!_consoleProvider.Shutdown && !Console.IsOutputRedirected)
```

Comment:
The reason the tests are failing because of this condition is that they redirect the input and send all the commands in the scripts (under src/SOS/SOS.UnitTests/Scripts).  You could try checking if the input is redirected:

```
 if (!_consoleProvider.Shutdown && (!Console.IsOutputRedirected || Console.IsInputRedirected))
```

Reply:
> they redirect the input and send all the commands in the scripts

Ah, that's the point. Thanks!
Locally it looks good with your suggestion.